dts/common/mcuboot.overlay

```diff
@@ -1,10 +0,0 @@
-/*
```

Comment:
Rather than removing mcuboot.overlay shouldn't we move it to all the sample applications which demonstrate MCUboot usage? If this commit is accepted the `zephyr,code-partition = &slot0_partition;` chosen property will always be defined even if we don't want to place the code at slot0_partition. There is also the following comment in each board dts file:
```
		/*
		 * If chosen's zephyr,code-partion is unset, the image will be
		 * linked into the entire flash device.  If it points to an
		 * individual partition, the code will be linked to, and
		 * restricted to that partition.
		 */
```

Reply:
I only found the comment in one file.  I've removed it now.  As for doing this in the application, I'm not against that, but would leave that to a cleanup after this PR gets merged.

Reply:
Should be addressed.

Reply:
Sorry for coming back late on this. The text is not always identical, however any board .dts file which defines `partitions` node will typically have something. E.g.
```
    /*
     * For more information, see:
     * http://docs.zephyrproject.org/latest/devices/dts/flash_partitions.html
     */
```

Reply:
Right, do you feel we need to update the flash_partition docs?

Reply:
Yes, I believe we should update the flash_partitions.html document. As for the comment within the board .dts files we can either:
- ensure that all comments refer to flash_partitions.html.
- remove all comments completely. Make sure that user is able to easily find flash_partitions.html document when searching/reading about MCUboot / bootloader support.

Also the help text of BOOTLOADER_MCUBOOT Kconfig option in ./Kconfig.zephyr needs to be updated. It currently states
> - Including dts/common/mcuboot.overlay when building the Device Tree in order to place and link the image at the slot0 offset

Reply:
@galak This PR is about to be merged. I just want to remind the the documentation update related to the modified way we link flash partitions doesn't seem to be done yet. Specifically the following paragraph

If the ``chosen`` node has no ``zephyr,code-partition`` property, the
application image link uses the entire flash device. If a
``zephyr,code-partition`` property is defined, the application link
will be restricted to that partition.

from `doc/devices/dts/flash_partitions.rst` file. Unless we do it later, after release freeze.

Reply:
Do you mind opening an issue for this, I don't want to hold up merging this on a doc update.  I'm happy to make the doc changes, just want to get this merged so it starts getting usage.

Reply:
OK, will do.